### PR TITLE
wip: typescript instead of prisma

### PR DESCRIPTION
**Pull request by DatGuyJonathan**
This is a copy-pasta from https://github.com/514-labs/moose/pull/549, minus doc changes & pr feedback changes. The goal is to get something merged to main in case we need to swarm on it.

I added an if-else to `parse_schema_file` so it can continue to work with current prisma schema. Next step is to iron out the wrinkles. Need to look into feedback from the original pr, and also what happens if we change `models.prisma` to `models.ts` with these interfaces.

```
interface UserActivity {
    eventId: string;
    timestamp: Date;
    userId: string;
    activity: string;
}

interface ParsedActivity {
    eventId: string;
    timestamp: Date;
    userId: string;
    activity: string;
}
```

### CODE REVIEW

File: apps/framework-cli/src/framework/schema.rs [hunk 1]
```diff
@@ -378,10 +408,152 @@ pub fn ast_mapper(ast: SchemaAst) -> Result<FileObjects, ParsingError> {
...
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsInterface(decl))) => {
+            ts_declarations.push(decl);
+        }
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsEnum(decl))) => {
+            enums.push(ts_enum_to_data_enum(decl));
+        }
+        _ => {}
+    });
+
+    println!(
```
Comment (callicles) [inline apps/framework-cli/src/framework/schema.rs]: Can we replace by `info!`?

File: apps/framework-cli/src/framework/schema.rs [hunk 2]
```diff
@@ -378,10 +408,152 @@ pub fn ast_mapper(ast: SchemaAst) -> Result<FileObjects, ParsingError> {
...
+        .filter_map(|field| match field {
+            swc_ecma_ast::TsTypeElement::TsPropertySignature(prop) => Some({
+                // match the key's sym if it's an ident
+                let name = match *prop.key.clone() {
+                    Expr::Ident(ident) => ident.sym.to_string(),
+                    _ => return None,
+                };
+                // match the type of the value and return the right column type
+                let TsTypeAnn { type_ann, .. } =
+                    *prop.type_ann.clone().expect("no type for property");
```
Comment (callicles) [inline apps/framework-cli/src/framework/schema.rs:471]: This is going to panic if the user fails to put a type. They will do that eventually, maybe we solidify this in a follow up PR with error handling.

**Review by callicles — APPROVED** [review]
Some small things - definitely some rough edges there but we should merge and iterate. nice work bringing this to the latest version

File: apps/framework-cli/src/framework/schema.rs (hunk 1 above)
Comment (DatGuyJonathan) [reply apps/framework-cli/src/framework/schema.rs]: Nice catch, will update it.

File: apps/framework-cli/src/framework/schema.rs (hunk 2 above)
Comment (DatGuyJonathan) [reply apps/framework-cli/src/framework/schema.rs:471]: Sounds good, I'll make a note of it.

**Pull request merged**